hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -200,6 +202,41 @@ public void run() {
         }, 1, 1, TimeUnit.SECONDS);
     }
 
+    private void openConnectionsToMembers() {
+        ClientClusterService clientClusterService = client.getClientClusterService();
+        Collection<Member> memberList = clientClusterService.getMemberList();
+        long startTimeMillis = System.currentTimeMillis();
+        while (true) {
+            Exception exception = null;
+            for (Member member : memberList) {
+                try {
+                    clientConnectionManager.getOrConnect(member.getAddress());
+                } catch (Exception e) {
+                    if (!clientClusterService.getMemberList().contains(member)) {
+                        //member left the cluster no need to connect
+                        continue;
+                    }
+                    exception = e;
+                }
+            }
+            if (exception == null) {
+                return;
+            }
+            ClientInvocationServiceImpl invocationService = (ClientInvocationServiceImpl) client.getInvocationService();
```

Comment:
these two lines can be put outside the loops since it does not change.

Reply:
I have put this intentionally after `if (exception == null) return` check so that in most of the case healthy run we do not pull the invocation service unnecessarily.

Reply:
No, I meant these lines:
            ClientInvocationServiceImpl invocationService = (ClientInvocationServiceImpl) client.getInvocationService();
 long invocationTimeoutMillis = invocationService.getInvocationTimeoutMillis();